services/rfq/relayer/inventory/manager.go

```diff
@@ -307,45 +304,78 @@ func (i *inventoryManagerImpl) ApproveAllTokens(ctx context.Context) error {
 			return fmt.Errorf("could not get chain client: %w", err)
 		}
 
-		for address, token := range tokenMap {
-			// approve RFQ contract.
+		for tokenAddr := range tokenMap {
 			// Note: in the case where submitter hasn't finished from last boot,
 			// this will double submit approvals unfortunately.
-			if address != chain.EthAddress && token.Allowances[contractRFQ].Cmp(big.NewInt(0)) == 0 {
-				tokenAddr := address // capture func literal
-				contractAddr, err := i.cfg.GetRFQAddress(chainID)
+			contractAddr, addrErr := i.cfg.GetRFQAddress(chainID)
+			if addrErr == nil {
+				err = i.approve(ctx, tokenAddr, contractAddr, backendClient)
 				if err != nil {
-					return fmt.Errorf("could not get RFQ address: %w", err)
+					return fmt.Errorf("could not approve RFQ contract: %w", err)
 				}
-				err = i.approve(ctx, tokenAddr, common.HexToAddress(contractAddr), backendClient)
+			}
+
+			contractAddr, addrErr = i.cfg.GetSynapseCCTPAddress(chainID)
+			if addrErr == nil {
+				err = i.approve(ctx, tokenAddr, contractAddr, backendClient)
 				if err != nil {
-					return fmt.Errorf("could not approve RFQ contract: %w", err)
+					return fmt.Errorf("could not approve SynapseCCTP contract: %w", err)
 				}
 			}
 
-			// approve SynapseCCTP contract
-			if address != chain.EthAddress && token.Allowances[contractSynapseCCTP].Cmp(big.NewInt(0)) == 0 {
-				tokenAddr := address // capture func literal
-				contractAddr, err := i.cfg.GetSynapseCCTPAddress(chainID)
+			contractAddr, addrErr = i.cfg.GetTokenMessengerAddress(chainID)
+			if addrErr == nil {
+				err = i.approve(ctx, tokenAddr, contractAddr, backendClient)
 				if err != nil {
-					return fmt.Errorf("could not get CCTP address: %w", err)
+					return fmt.Errorf("could not approve TokenMessenger contract: %w", err)
 				}
-				err = i.approve(ctx, tokenAddr, common.HexToAddress(contractAddr), backendClient)
+			}
+
+			parentAddr, addrErr := i.cfg.GetL1GatewayAddress(chainID)
+			if addrErr == nil {
+				contract, err := l1gateway.NewL1GatewayRouter(parentAddr, backendClient)
 				if err != nil {
-					return fmt.Errorf("could not approve SynapseCCTP contract: %w", err)
+					return fmt.Errorf("could not get L1Gateway contract: %w", err)
+				}
+				contractAddr, err := contract.EthGateway(&bind.CallOpts{Context: ctx})
+				if err != nil {
+					return fmt.Errorf("could not get L1ETHGateway address: %w", err)
+				}
+				err = i.approve(ctx, tokenAddr, contractAddr, backendClient)
+				if err != nil {
+					return fmt.Errorf("could not approve L1ETHGateway contract: %w", err)
+				}
```

Comment:
@dwasse Don't think we need to approve ETHGateway contract, only ERC20Gateway uses token approvals